yarn-project/sequencer-client/src/simulator/fake_public.ts

```diff
@@ -20,17 +35,31 @@ export class FakePublicCircuitSimulator implements PublicCircuitSimulator {
     const historicPublicDataTreeRoot = Fr.fromBuffer(publicDataTreeInfo.root);
 
     const execution = PublicExecution.fromTransactionRequest(this.db, tx, functionBytecode, portalAddress);
+
+    // Pad arrays to reach required length
     const result = await execution.run();
+    const args = tx.args;
```

Comment:
@sirasistant we had an issue running the acvm that it was complaining of arrays not being padded. So for example the args array only had 3 items instead of 8. I couldn't find if/where you were padding the arrays in the private client.